ndiego/social-sharing-block
# Add/local dev environment

I plan to suggest two PRs for the variations filter and had to set up a local dev environment to make changes. Therefore, I thought I might as suggest it as an PR for you and other contributors.

For convenience, I've added an easier local development environment using @wordpress/env. 
This allows for simpler setup of local dev environment and also ensures all contributors have the same local dev setup. More information about that package here: https://developer.wordpress.org/block-editor/reference-guides/packages/packages-env/
I've also added npm scripts to make it easier to use the wp-env version installed in the package. The naming of these scripts is arbitrary. 

I've also fixed the **node/npm version** used in this plugin. I guessed it was initially v14 (because all versions above try to migrate the package-lock.json). I then updated it to v16 to be on par with Gutenberg. This makes it easier for developers to use the correct node/npm version (using nvm).

**Testing:**
1. Make sure you have gut docker setup (see [@wordpress/env prerequisites](https://developer.wordpress.org/block-editor/reference-guides/packages/packages-env/#prerequisites))
2. Run `nvm use` if you have multiple node versions installed
3. Run `npm install`
4. Run `npm run dev:start` to startup local dev environment
5. Visit WordPress (probably on localhost:8888) and ensure the plugin is installed, active and usable.

Nothing about the plugin functionality or the build process changes/should change.
